docs(agents): record W&B product and platform facts for style passes (#2986)
wandb/docs: AGENTS.md

- **Company name**: W&B (not "Weights & Biases") in running text.
- **First mention pattern**: Use "W&B [Product]" on first mention of a product in a page, then drop "W&B" for later mentions — this is how Weave, Models, and Launch behave. Run, artifact, and report are looser: keeping "W&B" throughout (not just on first mention) is current usage too. Sweep is the exception that always drops it — see Sweeps below.
- **Product name capitalization**: Products stay capitalized on their own — W&B Weave → Weave, W&B Models → Models, W&B Launch → Launch. W&B artifact and W&B run are always lowercase ("an artifact", "a run"), matching how the API names these objects.
- **The web app is "W&B"** ("Weights & Biases" in full) — not the "W&B App". Now that a mobile app exists, "app" no longer identifies the web product. Name the interface itself the **W&B UI**, a component of W&B, when a sentence needs to distinguish it from the API or SDK.
- **W&B Python SDK**: The `wandb` package is the W&B Python SDK. Use "the W&B Python SDK (`wandb`)" as the full form and "the `wandb` library" as the short form.

> - **Weave UI labels**: When one of these words names a tab, view, page, table, or panel in the Weave UI, it keeps its UI casing: Traces, Calls, Threads, Conversations, Spans, Signals, Agents, Scorers, Monitors, Evals, Models, Datasets, Prompts, Assets, Dashboard, Playground, Leaderboard. The same words as ordinary nouns in prose ("the calls in this trace") are lowercase throughout the docset — don't recapitalize them.

Reviewer: ```suggestion
- **Weave UI labels**: When one of these words names a tab, view, page, table, or panel in the Weave UI, it keeps its UI casing: Traces, Calls, Threads, Conversations, Spans, Signals, Agents, Scorers, Monitors, Evals, Models, Datasets, Prompts, Assets, Dashboard, Playground, Leaderboard. The same words as ordinary nouns in prose ("the spans in the trajectory") are lowercase throughout the docset — don't recapitalize them.
```

> - **Project navigation**: The navigation inside a W&B project is the **project sidebar** (the **Weave project sidebar** in Weave docs), not the "left sidebar", "sidebar menu", or "left navigation". Bare "sidebar" is fine. This covers W&B's own in-project navigation only — leave third-party console steps (for example, the Microsoft Entra ID and Slack procedures under `platform/hosting/`), W&B organization- and team-level navigation, and `release-notes/` wording as written.

Reviewer: ```suggestion
- **Project navigation**: The navigation inside a W&B project is the **project sidebar** , not the "left sidebar", "sidebar menu", or "left navigation". Bare "sidebar" is fine. This covers W&B's own in-project navigation only — leave third-party console steps (for example, the Microsoft Entra ID and Slack procedures under `platform/hosting/`), W&B organization- and team-level navigation, and `release-notes/` wording as written.
```

> - **Project navigation**: The navigation inside a W&B project is the **project sidebar** (the **Weave project sidebar** in Weave docs), not the "left sidebar", "sidebar menu", or "left navigation". Bare "sidebar" is fine. This covers W&B's own in-project navigation only — leave third-party console steps (for example, the Microsoft Entra ID and Slack procedures under `platform/hosting/`), W&B organization- and team-level navigation, and `release-notes/` wording as written.

Reviewer: ```suggestion
- **Project navigation**: The navigation inside a W&B project is the **project sidebar** , not the "left sidebar", "sidebar menu", or "left navigation". Bare "sidebar" is fine. This covers W&B's own in-project navigation only — leave third-party console steps (for example, the Microsoft Entra ID and Slack procedures under `platform/hosting/`), W&B organization- and team-level navigation, and `release-notes/` wording as written.
```
